packages/server/src/fhir/operations/plandefinitionapply.ts

```diff
@@ -119,6 +123,53 @@ async function createQuestionnaireTask(
   ]);
 }
 
+/**
+ * Creates a Task and RequestGroup action to request a resource.
+ * @param repo - The repository configured for the current user.
+ * @param requester - The user who requested the plan definition.
+ * @param subject - The subject of the plan definition.
+ * @param action - The PlanDefinition action.
+ * @param encounter - Optional encounter reference.
+ * @returns The RequestGroup action.
+ */
+async function createActivityDefinitionTask(
+  repo: Repository,
+  requester: Reference<Bot | ClientApplication | ProfileResource>,
+  subject: Reference<Patient>,
+  action: PlanDefinitionAction,
+  encounter: Reference<Encounter> | undefined
+): Promise<RequestGroupAction> {
+  const activityDefinition = await repo.readReference<ActivityDefinition>({ reference: action.definitionCanonical });
+  switch (activityDefinition.kind) {
+    case 'ServiceRequest': {
+      const serviceRequest = await repo.createResource({
+        resourceType: 'ServiceRequest',
+        status: 'draft',
+        intent: activityDefinition.intent as ServiceRequest['intent'],
+        subject: subject,
+        requester: requester as ServiceRequest['requester'],
+        encounter: encounter,
+        code: activityDefinition.code,
+      });
+
+      return createTask(repo, requester, subject, action, encounter, [
+        {
+          type: {
+            text: 'ServiceRequest',
+          },
+          valueReference: {
+            display: action.title,
+            reference: getReferenceString(serviceRequest),
+          },
+        },
+      ]);
```

Comment:
I don't think we need to create a `Task` for the `ServiceRequest` - the `ServiceRequest` should be viable on its own.

```suggestion
      return {
        title: action.title,
        resource: createReference(serviceRequest),
      };
```

Reply:
The idea behind it is to track status similar to other tasks, add notes, due date, etc.
![Screenshot 2025-03-17 at 6 17 05 PM](https://github.com/user-attachments/assets/25cf3ea5-2067-49a5-89c2-71e75b4f8b18)

Reply:
and it can record the output. e.g  "procedure"

Reply:
Got it.  Claude says this is ok ✅